### ISSUE TITLE
In the advanced search, add a "tags" field to the translation side (right side)

### ISSUE THREAD

**Issue by ckjpn** [Tatoeba/tatoeba2#2143]
https://tatoeba.org/eng/sentences/advanced_search

## The problem

It's not easy to find French or other language sentences related to travel and other topics.

See this Wall post.

https://tatoeba.org/eng/wall/show_message/34172#message_34172

## One possible solution

Add the "tags" field, just like the "owner" field on the Translations side of the search.

**Comment by jiru** [comment]
Isn’t it already possible to find French sentences about travel by looking up related words in French like hôtel, séjour, voyage etc.? What’s the point of searching in English to find French sentences?

**Comment by ckjpn** [comment]
> What’s the point of searching in English to find French sentences?

1. Not all French sentences are tagged.
2. This suggestion wouldn't limit finding tags on English sentences, but sentences in any sentences with tags.
3. If someone has taken the time to tag sentences in other languages, wouldn't it be helpful if those tags could also help people find related sentences in other languages.

Here is the first example, that I gave in the Wall post, but this way requires extra steps to translate the French into another language, as explained in the Wall post.

https://tatoeba.org/sentences/search?trans_filter=limit&trans_to=fra&tags=hotel

**Comment by agrodet** [comment]
I think I don't understand CK's needs but I think @Guybrush88 expressed a similar wish in #1333

**Comment by ckjpn** [comment]
The enhancement would be to allow people to find sentences in a language that doesn't have many tags, that are tagged in languages that are linked to them.

In other words, if someone wanted to find French sentences related to hotels, they could use search for translations that were tagged "hotel", on the right side of the advanced search.

Ricardo14, in this Wall post https://tatoeba.org/eng/wall/show_message/34172#!#message_34172, wanted French sentences related to traveling (hotel, etc.).

If he the advanced search added the option as described, he could easily find and translate such French sentences.  Of course, perhaps not all results would be related to the tag of the other languages, but many of them would be, I think.

**Comment by ckjpn** [comment]
On a related note, having all of the options on the left side of the advanced search page, also on the right side of the page, would be useful.  Many are already there. However, Tags, Lists and "Owned by a self-identified native" aren't there yet.

Adding Lists to that side of the page would allow this kind of search.

- Find French sentences linked to English sentences linked to more than 9 languages (List 7055).
- Find French sentences linked to English sentences linked to more than 9 languages (List 7055), not yet translated into one's own native language.

**Comment by jiru** [comment]
About the original issue that is to find French sentences on the topic of travel. I think that looking up French words is still a better than your suggested solution, because not every French sentence is translated in English (and never will), and because tags are not covering all sentences (and never will).

> allow people to find sentences in a language that doesn't have many tags

About that other problem, another way to solve it could be to have tags that apply to a whole sentence group instead of a single sentence.

**Comment by RyckRichards** [comment]
Actually, I'd like to share my point of view since I was the one who requested that list.

Not everyone speaks English or French. All of us are aware of that. If I'm not wrong, there are more English speakers than French speakers and sometimes, people can express themselves better in English rather in French.

That said, just few people would imagine sentences which include the words mentioned above "hôtel", "séjour". It made me ask on the Wall sentences that would include these words (and much more, for sure) to actually learn from them and tell everybody I'm inviting to Tatoeba - perhaps you want to study something about traveling and there are a couple of sentences in this list. It'd be awesome if you can translate them in your native and strongest languages.

Would tags solve the problem? Not sure. Perhaps we'd gotta have a kind of system that automatically tags the translations if any of them were translated.

example

I love Paris. << tagged as "city"
Eu amo Paris. << *automatically* tagged as "city"
Me gusta Paris. << *automatically* tagged as "city"

Please, let me know if I was not clear enough :)

**Comment by agrodet** [comment]
Automatic tagging of translations is impossible because we cannot be sure the sentence was not translated to talk about a different topic. In your example, Paris could be Paris Hilton, who is a person, not a city. Automatic suggestion of tags, however, could be useful, but that's a whole different story.

I still don't understand your issue. From what you explained, I thought that jiru answers above would answer your request. Are these solutions lacking something? Can you imagine another solution than having useful sentences tagged and simply looking for them in French?

**Comment by RyckRichards** [comment]
Not for now, to be honest. People could add sentences about traveling to that list... Can't imagine nothing now but I'll think about something.

The biggest issue is: How to gather sentences which someone requested about a specific topic (Traveling, sports, house stuff and so on)

In the case you've mentioned, if someone tags one sentence which contains "Paris" and it's not about the city (imagining that this feature was implemented), someone else would remove this tag (and so, from all sentences). 

Tags is just an approach.

**Comment by ckjpn** [comment]
> Automatic tagging of translations ....

I agree this is a bad idea. Even if the idea is just to transfer tags from one language to another, this would be a bad idea. 

However, as I mentioned before, a good reason for having tags as an option in the translations is that someone like RyckRichards could easily find sentences in French shown in search results that may be related to "hotel" or whatever kind of sentence he is looking for that may be already tagged in another linked language.

One example of this is the Russian sentence that is tagged "hotel" that is linked to a French sentence.

People looking for sentences related to hotels would not naturally think of searching for the word "room," and even if they did, there would be too many sentences with "room" that would be unrelated to "hotel."

rus
Ваш номер готов.
eng
Your room's ready.
fra
Votre chambre est prête.
fra
Votre chambre est prête, maintenant.

The fact that some sentences in French may not be related to the tags of the other sentences would possibly mean that not all sentences found would be related, but many of them would, so this would still be a valid approach for searching for such sentences.

**Comment by ckjpn** [comment]
Perhaps this would be a better name for this issue.

**Add some options that are on the left side (Sentences) of the advanced search function to the right side (Translations), too.**

At this time, it would be these options.

**Tags:**

**Owned by self-identified native**

**Belongs to list:**

However, if in the future other options are added, adding the option to both sides should be considered.

**Question:**
Would it be better to open this as a new issue?


Link to the advanced search: https://tatoeba.org/eng/sentences/advanced_search

**Comment by jiru** [comment]
> Perhaps this would be a better name for this issue.
> **Add some options that are on the left side (Sentences) of the advanced search function to the right side (Translations), too.**

It’s better to name it after the problem, not the solution.

> Would it be better to open this as a new issue?

Yes, but please try to explain the problem you are trying to solve.

> The biggest issue is: How to gather sentences which someone requested about a specific topic (Traveling, sports, house stuff and so on)

Thanks for clarifying the use case.

The idea of transferring tags to linked sentences is just one way to achieve this goal, but I like it because it also helps corpus categorization in general. I can see that it has advantages (the information conveyed by the tag spreads across languages) but also drawbacks (we can’t be sure the tag applies to another language/culture or in a specific context).

Maybe the solution is in between: assisting contributors in the transfer of tags from one sentence to another linked sentence.

Maybe we could split tags into two categories: generic and language-specific.  "travel" would be generic, while e.g. "passato prossimo" would be language-specific. Only the travel tag would "spread" to other languages while passato prossimo wouldn’t. It’s difficult however to judge how generic a given tag can be. Maybe there is a language where the concept of travel doesn’t exist or is assimilated with another concept?

Or maybe (and that’s related to #54) we could make all tags language-specific and have a kind of linking feature between tags to control spreading: if an English tag "travel" is linked with the French tag "voyage", then the tag automatically transfers between French and English. But we should still be able add or remove tags independently (for example if French contributors enventually disagree that a particular sentence should be tagged as "voyage"). Such linking could be eventually used to navigate users from the English tag page "travel" to the French tag page "voyage".
